<a href="https://colab.research.google.com/github/juanfelipegonzalezalfonso/programacion1/blob/main/Parcial_2260663.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# CREAR MODULOS Y PAQUETES:

import os
os.makedirs("op", exist_ok=True)

%%writefile op/operaciones.py



#Juan Felipe Gonzalez Alfonso
codigo: 2260663

#Instruccion 1

In [1]:
import numpy as np
import pandas as pd

In [11]:
np.random.seed(0)
datos = np.random.randint(130, 196, 50)
datos

array([174, 177, 194, 139, 151, 166, 142, 188, 195, 169, 176, 167, 155,
       139, 150, 177, 194, 179, 159, 149, 149, 144, 169, 162, 195, 139,
       187, 162, 161, 153, 165, 185, 158, 164, 130, 130, 166, 183, 135,
       168, 147, 134, 172, 188, 161, 131, 195, 171, 187, 165])

In [12]:
df = pd.DataFrame({
    "datos": datos
})
df

,datos
0,174
1,177
2,194
3,139
4,151
5,166
6,142
7,188
8,195
9,169


In [14]:
df.loc[df["datos"]>150]

,datos
0,174
1,177
2,194
4,151
5,166
7,188
8,195
9,169
10,176
11,167


In [15]:
df["datos"].apply(lambda x: (x - 163.92)/19.162932)

,datos
0,0.526016
1,0.682568
2,1.569697
3,-1.300427
4,-0.674218
5,0.108543
6,-1.143875
7,1.256593
8,1.621881
9,0.265095


In [28]:
df.describe()

,datos
count,50.000000
mean,163.920000
std,19.162932
min,130.000000
25%,149.250000
50%,165.000000
75%,177.000000
max,195.000000


#el promedio es 163.92, la desviacion estandar de los datos es 19.162932
#el valor minimo es 130 y el maximo es 195

#Instruccion 2

In [96]:
df1 = pd.DataFrame({
    "Nombre": ["ricardo", "pablo", "enrique", "juan", "pedro"],
    "Edad": [22, 24, 30, 35, 18],
    "ingreso": [1000000, 200000, 3500000, 280000, 453000],
    "Pais": ["Venezuela", "Colombia", "Colombia", "Venezuela", "Japon"]
})

1

In [83]:
df1.head(1)

,Nombre,Edad,ingreso,Pais
0,ricardo,22,1000000,venezuela


2

In [84]:
df1['Nombre']

,Nombre
0,ricardo
1,pablo
2,enrique
3,juan
4,pedro


3

In [85]:
df1.loc[df1["Edad"] > 23]

,Nombre,Edad,ingreso,Pais
1,pablo,24,200000,Colombia
2,enrique,30,3500000,Colombia
3,juan,35,280000,Venezuela


4

In [86]:
cambio = df1['Edad'].apply(lambda x: x+2)
cambio

,Edad
0,24
1,26
2,32
3,37
4,20


5

In [97]:
grupo = df1.groupby("Pais")["ingreso"].mean()
grupo

,ingreso
Pais,
Colombia,1850000.0
Japon,453000.0
Venezuela,640000.0


6

#Instruccion 3

In [78]:
salon1 = np.random.normal(loc=7.2, scale=0.4, size=58)


In [79]:
salon2 = np.random.gamma(shape=306, scale=0.0229, size=58)

In [80]:
df2 = pd.DataFrame({
    "salon 1": salon1,
    "salon 2": salon2
})

,salon 1,salon 2
0,7.067624,6.828539
1,7.005914,6.985332
2,7.175915,6.525770
3,6.576141,7.003562
4,7.083438,6.752986
5,7.521629,7.119453
6,7.494331,6.707947
7,7.118944,7.613854
8,7.316099,7.100243
9,6.858073,7.250659


In [81]:
df2.describe()

,salon 1,salon 2
count,58.000000,58.000000
mean,7.270847,6.968097
std,0.430047,0.413559
min,6.268519,5.963593
25%,7.030047,6.689382
50%,7.218542,6.959590
75%,7.566401,7.282300
max,8.325098,7.751763


la media(promedio) del salon 1 se encuentra en 7.27, la desviacion estandar es de 0.43, el valor minimo es de 6.26, su maximo es de 8.32 , la mediana del primero es de 7.21, cuantil 25% es 7.03 y 75% de 7.56

la media(promedio) del salon 2 se encuentra en 6.96, la desviacion estandar es de 0.41, el valor minimo es de 5.96, su maximo es de 7.75 , la mediana del segundo es de 6.95, cuantil 25% es 6.68 y 75% de 7.28


#4

In [89]:
from scipy import stats
from scipy.stats import shapiro
shapiro(df2["salon 1"])

ShapiroResult(statistic=np.float64(0.9863598456464102), pvalue=np.float64(0.7586232273214917))

In [90]:
from scipy import stats
from scipy.stats import shapiro
shapiro(df2["salon 2"])

ShapiroResult(statistic=np.float64(0.9870616707368133), pvalue=np.float64(0.7926118090652816))

en ambos salones el valor de p es mayor que 0.5, se cumple el suspuesto de normalidad

In [100]:
stats.ttest_rel(df2["salon 1"], df2["salon 2"])

TtestResult(statistic=np.float64(3.6779502130699595), pvalue=np.float64(0.0005223971223511537), df=np.int64(57))

no se rechaza porque el porcentaje de desconfianza es del 3.67

In [104]:
stats.kendalltau(df2["salon 1"], df2["salon 2"])

SignificanceResult(statistic=np.float64(-0.055051421657592255), pvalue=np.float64(0.5415791729603694))

no estan coorelacionados porque el valor de p es mayor a 0.5

In [108]:
stats.spearmanr(df2["salon 1"], df2["salon 2"])

SignificanceResult(statistic=np.float64(-0.08296164139161463), pvalue=np.float64(0.5358294634938489))

no estan coorelacionados porque el valor de p es mayor a 0.5

In [111]:
import os
os.makedirs("comparacion", exist_ok=True)


In [112]:
%%writefile comparacion/pruebat.py

stats.ttest_rel(df2["salon 1"], df2["salon 2"])

Writing comparacion/pruebat.py


In [ ]:
%%writefile comparacion/cor.py

